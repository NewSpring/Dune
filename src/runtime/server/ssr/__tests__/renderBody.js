import React from "react";
import { Route } from "react-router";
import render from "../renderBody";

const Home = () => <h1>Home</h1>;
const Hello = () => <h3>Hello World</h3>;
const App = () => (
  <div>
    <Route exact path="/" component={Home} />
    <Route exact path="/hello" component={Hello} />
  </div>
);

it("renders the body of the app into a static router context", () => {
  const ctx = {};
  const context = {};
  const result = render({
    ctx,
    context,
    path: "/",
    App,
  });

  expect(result.body).toMatchSnapshot();
});

it("renders the body of the app into a static router context based on route", () => {
  const ctx = {};
  const context = {};
  const result = render({
    ctx,
    context,
    path: "/hello",
    App,
  });

  expect(result.body).toMatchSnapshot();
});
