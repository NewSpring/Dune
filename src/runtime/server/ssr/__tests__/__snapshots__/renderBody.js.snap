// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the body of the app into a static router context 1`] = `"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"1494228789\\"><h1 data-reactid=\\"2\\">Home</h1><!-- react-empty: 3 --></div>"`;

exports[`renders the body of the app into a static router context based on route 1`] = `"<div data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"818356684\\"><!-- react-empty: 2 --><h3 data-reactid=\\"3\\">Hello World</h3></div>"`;
